src/store/LocalStorageStore.ts

```diff
@@ -0,0 +1,207 @@
+import {CommentData, EntityType, FileData, ProjectData, SectionData} from '../types';
+
+export interface LocalStorageDependency {
+    getItem(key: string): string | null;
+    setItem(key: string, value: string): void;
+    clear(): void;
+}
+
+// TODO: versioning of the store allows for migrations
+// const LOCAL_STORAGE_KEY_VERSION = 'version';
+
+const LOCAL_STORAGE_KEY_PROJECTS = 'projects';
+const LOCAL_STORAGE_KEY_SECTIONS = 'sections';
+const LOCAL_STORAGE_KEY_FILES = 'files';
+const LOCAL_STORAGE_KEY_COMMENTS = 'comments';
+
+export type StoreData = {
+    projects: ProjectData[];
+    sections: SectionData[];
+    files: FileData[];
+    comments: CommentData[];
+}
+
+export class LocalStorageStore {
+    private ls: LocalStorageDependency;
+
+    private currentData: StoreData;
+
+    constructor(ls: LocalStorageDependency) {
+        this.ls = ls;
+        this.currentData = this.migrateLocalStorageStore();
```

Comment:
This file should only need to be edited if we add more types of data to the store